WoT-stats/src/app/components/layout/layout.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component } from '@angular/core';
+import { RouterOutlet } from '@angular/router';
+import { RouterModule } from '@angular/router';
+import { MatToolbarModule } from '@angular/material/toolbar';
+import { MatSidenavModule } from '@angular/material/sidenav';
+import { MatIconModule } from '@angular/material/icon';
+import { MatListModule } from '@angular/material/list';
+import { MatButtonModule } from '@angular/material/button';
+import { AuthService } from '../../service/auth/auth.service';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-layout',
+  standalone: true,
+  imports: [
+    RouterOutlet,
+    RouterModule,
+    MatToolbarModule,
+    MatSidenavModule,
+    MatIconModule,
+    MatListModule,
+    MatButtonModule
+  ],
+  templateUrl: './layout.component.html',
+  styleUrl: './layout.component.scss'
+})
+export class LayoutComponent {
+
+  accessToken: string = '';
+
+  constructor(private authService: AuthService,   private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.route.queryParams.subscribe(params => {
+      this.accessToken = params['access_token'];
```

Comment:
The better way is to set an access Token to localStorage or your service for auth. It's a bad idea to have it in your component because when you create another page with a different route you will not have access to this token.